src/k2/module/k23si/client/k23si_client.h

```diff
@@ -59,15 +59,10 @@ class K2TxnOptions{
 template<typename ValueType>
 class ReadResult {
 public:
-    ReadResult(Status s, dto::K23SIReadResponse<ValueType>&& r) : status(std::move(s)), response(std::move(r)) {}
```

Comment:
not sure why you removed this constructor - i think it is still usable when you don't want to construct a ValueType twice.